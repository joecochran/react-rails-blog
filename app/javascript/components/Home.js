import React, { Component } from 'react';

class Home extends Component {
  componentDidMount() {
    document.title = 'Dashboard';
  }

  render() {
    return (
      <div className="container mx-auto">
        <h1 className="mb-6 mt-6">Dashboard</h1>
      </div>
    );
  }
}

export default Home;
