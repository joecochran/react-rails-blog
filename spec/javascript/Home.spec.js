import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Home from 'components/Home';

configure({ adapter: new Adapter() });

describe('home component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
