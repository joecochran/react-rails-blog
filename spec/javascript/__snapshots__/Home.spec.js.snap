// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        className="mb-6 mt-6"
      >
        Dashboard
      </h1>,
      "className": "container mx-auto",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Dashboard",
        "className": "mb-6 mt-6",
      },
      "ref": null,
      "rendered": "Dashboard",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="mb-6 mt-6"
        >
          Dashboard
        </h1>,
        "className": "container mx-auto",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Dashboard",
          "className": "mb-6 mt-6",
        },
        "ref": null,
        "rendered": "Dashboard",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
